### ISSUE TITLE
Should loading event data no longer create weights with variances?

### ISSUE THREAD

**Issue by SimonHeybrock** [scipp/scippnexus#91]
After scipp/scipp#2895, broadcasting of variances will no longer be supported. This implies that the "standard" paradigm of handling uncertainties will not be feasible any more. This further implies that there is limited use to carrying uncertainties of "counts", since uncertainties can simply be computed later on.

We should therefore consider avoiding the overhead of creating variances for the weights when an `NXdetector` with `NXevent_data` is loaded. This would save both memory and compute resources.

**Comment by nvaytet** [comment]
I think having special handling of counts units is not a good idea, that's how we ended up doing strange things with the resampling mechanism.
As you say, the variances on counts are just something we set by hand, when loading the counts data, it is not there in the data file. And they can be computed later.

Someone, not in neutron science, might want to have counts without variances, why should they suddenly see errorbars on their plot?
Which is probably why you mention a convenience method, but I think the convenience method should just be
```Py
data.variances = data.data
```
